slash_commands/adora-sac.js

```diff
@@ -0,0 +1,168 @@
+const { SlashCommandBuilder } = require('discord.js');
+const { isTower, allTowers, allUpgradeCrosspathSets, costOfTowerUpgrade, isTowerUpgrade, cumulativeTowerUpgradePathCosts } = require('../helpers/towers.js');
+const { red, cyber } = require('../jsons/colors.json');
+
+const XP_CAP = 344740;
+const XP_CAP_ADJUSTED = Math.ceil(XP_CAP / (4*5));
+
+// deprioritize water towers, illegal towers in CHIMPS, towers that are weird to sac in 2mpc, etc.
+const TOWER_PRIORITIES = new Map();
+TOWER_PRIORITIES.set('banana_farm', -1000); // illegal in chimps
+TOWER_PRIORITIES.set('monkey_sub', -700); // water tower
+TOWER_PRIORITIES.set('monkey_buccaneer', -850); // water tower
+TOWER_PRIORITIES.set('beast_handler', -620); // water tower in some paths
+TOWER_PRIORITIES.set('heli_pilot', -500); // can't be placed on midnight mansion
+TOWER_PRIORITIES.set('monkey_ace', -380); // weird in 2mp
+TOWER_PRIORITIES.set('sniper_monkey', -250); // weird in 2mp
+TOWER_PRIORITIES.set('druid_monkey', -120); // jungle druid is weird in 2mp
```

Comment:
![image](https://github.com/hemisemidemipresent/cyberquincy/assets/68867396/64c3c204-3d81-4ebb-a1ec-1715effaf1a9)

Reply:
@Landmaster does this mean that deprioritization isn't working as expected? I see sniper is deprioritized, but banana farm is much more so.

Reply:
I calculated that by removing banana farms, every price point is still hit, so banana farms should never really show up, whereas they show up close to 1,000 times now.

Reply:
Oh right I forgot though that the price point might have taken more towers to hit in such cases..

Reply:
> im guessing its that while sniper is deprioritized, farm is excluded so this is intended behavior

My point is that without excluded towers (the first command execution), the second execution should have showed up because sniper is preferred over banana farm.

Reply:
Ok, so I reminded myself _exactly_ how tower de-prioritization works. Basically the price values are calculated first, then the tower tiebreaks come after. Basically the algorithm reached the banana farm cost, then saw that only banana farm could satisfy that cost, rather than weighing the two different sets of costs based on the towers they map to.

Not sure if that's something that's easily remediable...regardless, users can utilize excluded towers.